### DISCUSSION TITLE
Author profile content as php clone into a module?

### DISCUSSION THREAD

**Discussion by ApfailKuchen** [templaza/astroid-framework#658] [Q&A]
Hello, my dears :)

I recently reported that I had problems with navigation because an external extension was integrated underneath.

I was able to solve the problem. However, what I failed to consider was the mobile view.. (shame on my head)

Now this problem could be solved by loading the author box within a module instead of within the post and next to the extension underneath. (Because here we have the error in the mobile view that it makes the page much wider in portrait format)

Is it possible to bring the code of the author box into a module? I have tried something but land by

Call to a member function renderAuthorInfo() on null

or

Call to protected method

I would be very grateful for help

**Comment by sonvnn** [comment]
@ApfailKuchen Can you screenshot mobile issue? So I can give you some advise for it.

**Reply by ApfailKuchen** [reply]
Hey hey,

If you want, you can view the issue on the cloned page by resizing your browser window.

https://clone.apfailkuchen.de/android/test

User: Test3r_admin
Passwort: Pic0ly_rE3T!_Bh$61

However, I agree that the error can be seen even better on the cell phone.

![Issue](https://github.com/templaza/astroid-framework/assets/84144100/382dc750-8182-418d-836d-1360bee7efa0)

**Reply by ApfailKuchen** [reply]
The problem is that the extension for the rating and the author box share the same space.

and therefore I would like to outsource the code for the author box and put it in a module

----
Please excuse me, but unfortunately I have to go to bed now 🙈 (GMT+2 / 4:30 am)

**Reply by ApfailKuchen** [reply]
It's regrettable, I was hoping that with your experience you might have some advice on how to solve this problem. Still, thank you for taking a look. ❤️

**Comment by ApfailKuchen** [comment]
Hello dear interested helper,

I'm still looking for a way to correct the incorrect display of the author box and the rating extension.

Contrary to the incorrect css label, this is a plugin story.

One idea for implementation (solving the problem) would be to offer the template another place to insert the author box.

A class/position like:
onContentAfterTitle
onContentBeforeDisplay
onContentAfterDisplay

i need someone that show me a way to  add another class/position here like for example...

"onContentSecondAfterDisplay"

 ... which means that the author box can be displayed last, both things would have separate positions in the template and would not be intertwined.

I would be so grateful if anyone had a solution or advice on this

**Comment by Chacapamac** [comment]
My browser say that it’s risky to open the link and I’m not sure of what exactly is the problem?

**Reply by ApfailKuchen** [reply]
Hello Chacapamac, nice to see you again, I hope you are well? ❤️

The browser warning can be ignored. It's because I shared the URL above with http**S** but the cloned site doesn't use SSL.
Sorry, that's my fault. 

Just use http://clone.apfailkuchen.de/android/test and everything is fine.

Please excuse my English, I'll try to explain as best as I can.

Do you remember my first post here?

It was about an extension not being integrated correctly.
The site navigation was higher than the extension (and right next to it) and the author box .

I was able to fix this so that the navigation is now the last point in the article view.

By chance it turned out that the integration point (onContentAfterDisplay) of Joomla or Astroid is somehow the same as that used for the author box. Therefore, the author box and the extension are directly opposite each other (next to each other). 

This situation looks really good in the desktop view. 
![Bild_2024-05-03_181713631](https://github.com/templaza/astroid-framework/assets/84144100/91380a85-cb5b-4741-8a68-303f287b55ce)

However, I didn't think about the responsive design. This is where the design issue occurs as soon as the cell phone is in portrait orientation.

![Bild_2024-05-03_183158892](https://github.com/templaza/astroid-framework/assets/84144100/aae97c6f-3354-4215-b895-d732463010cc)

Because Astroid / Joomla uses the same integration point (onContentAfterDisplay) for both things, both things are also displayed next to each other in the mobile view. This circumstance then expands the width of the website.

To solve this problem, I thought of finding a way to tell the framework to use a new integration point for the author box, one that comes after onContentAfterDisplay, so that the extension and the author box are under each other instead of next to each other.

Unfortunately, I don't know how to implement this.

I hope anyone can understand what I want to say and wish everyone who reads this a nice weekend. :-)

[com_content_article_default(clonedsite).zip](https://github.com/templaza/astroid-framework/files/15202802/com_content_article_default.clonedsite.zip)

**Comment by ApfailKuchen** [comment]
Why did the topic receive the “Not Astroid related” label?

The first lines of `com_content/article_default.php`, which control the integration of extensions, say that they belong to Astroid?!

```
/**
 * @package   Astroid Framework
 * @author    Astroid Framework https://astroidframe.work
 * @copyright Copyright (C) 2023 AstroidFrame.work.
 * @license https://www.gnu.org/licenses/gpl-2.0.html GNU/GPLv2 or Later
```

When Astroid implements extensions and places them next to each other instead of one below the other then this is a relevant mistake by Astroid I would say 👀
